examples/step-67/doc/intro.dox

```diff
@@ -0,0 +1,686 @@
+in vector form with separate shape functions for the different components and
+letting $w_i(t)$ go through the density, momentum, and energy variables,
+respectively. In this form, the space dependence is contained in the shape
+functions and the time dependence in the unknown coefficients $w_i$. As
+opposed to the continuous finite element where some shape functions span
+across element boundaries, the shape functions are local to a single element
+in DG methods, with a discontinuity from one element to the next. The
+continuity of the solution is instead imposed weakly by the numerical fluxes
+specified below. This allows for some additional flexibility, for example to
+introduce directionality in the numerical method by e.g.~upwinding.
+
+DG methods are popular methods for solving problems of transport character
+because they combine low dispersion errors with controllable dissipation on
+barely resolved scales. This makes them particularly attractive for simulation
+in the field of fluid dynamics where a wide range of active scales needs to be
+represented and the pollution of the well-resolved features by inadequately
+resolved ones is undesired. Furthermore, high-order DG methods are well-suited
+for modern hardware with the right implementation. At the same time, DG
+methods are no silver bullet. In particular when the solution develops
+discontinuities (shocks), as is typical for the Euler equations in some flow
+regimes, high-order DG methods tend to oscillatory solutions, like all
+(unlimited) high-order methods. This is a consequence of Godunovs theorem that
+states that any total variation limited (TVD) scheme that is linear (like a
+basic DG discretization) can at most be first-order accurate. Put differently,
+since DG methods aim for higher order accuracy, they cannot be TVD on
+solutions that develop shocks. Even though some communities claim that the
+numerical flux in DG methods can control dissipations, this is of limited
+value unless <b>all</b> shocks in a problem align with cell boundaries. Any
+shock that passes through the interior of cells will again produce oscillatory
+components due to the high-order polynomials. In the finite element and DG
+communities, there exist a number of different approaches to deal with shocks,
+for example the introduction of artificial diffusion on troubled cells (using
+a troubled-cell indicator based e.g. on a modal decomposition of the
+solution), a switch to dissipative low-order finite volume methods on a
+subgrid, or the addition of some limiting procedures. Given the ample
+possibilities in this context, combined with the considerable implementation
+effort, we refrain from the regime of the Euler equations with pronounced
+shocks, and rather concentrate on the regime of subsonic flows with wave-like
+phenomena. For a method that works well with shocks (but is more expensive per
+unknown), we refer to the step-69 tutorial program.
```

Comment:
I guess it will be quite hard to match the per-dof performance you get with the plain DG code as soon as you start limiting, no matter what scheme you choose. Maybe artificial viscosity would be fastest in terms of DoF/s, but that is not nearly as good in terms of solution quality/crispness as the second-order variant of the approach from step-69.